app/scripts/lib/app-start.js

```diff
@@ -243,15 +247,48 @@ function (
       this._metrics.init();
     },
 
-    initializeIframeChannel: function () {
-      if (this._isIframe()) {
-        this._iframeChannel = new IframeChannel();
-        this._iframeChannel.initialize({
-          window: this._window,
-          origin: this._relier.get('origin'),
-          metrics: this._metrics
-        });
+    _getExpectedIframeOrigin: function () {
+      if (! this._isInAnIframe()) {
+        return null;
       }
+      if (this._isFxDesktop()) {
+        // If in an iframe for sync, the origin is
+        // checked against a pre-defined origin sent
+        // from the server.
+        return this._config.firstRunOrigin;
```

Comment:
Only allowing a single server will make it very difficult for the growth team to test this flow on their test servers. A more ideal situation is to fetch a list of allowed origins. Or, we possibly use a hybrid approach - register each individual growth team server with the oauth server. Each growth team server would have an individual client_id. When they load FxA in the first run flow, they pass in a client_id. We fetch the information for the specified client_id, but still use the Sync flow.